### PR TITLE
Add greedy algo and send error message

### PR DESCRIPTION
**Pull request by szhuang1**
1. Add the greedy algorithm as a method to decompose the NEgraph.
2. Add an error message to the final output of the clique_test function if for some focal assignments in the biclique that contains Zobs, exposures for each unit are the same.
3. Slightly modified the given greedy algorithm to get the true decomposition of NEgraph. Specifically, there are only three cases where we need to consider that the input ne is not binary, and I change ne to ne^2 in these three places:
- the end of function get_clique_size, 
- the end of get_clique,
- calculating units_leftover. 
Also, when calculating zobs_there, focalass should equal Zobs_id rather than "1".
4. Update variations in selecting decomposition algorithms in the two examples.

### CODE REVIEW

File: R/functions.R [hunk 1]
```diff
@@ -103,8 +123,18 @@ clique_test = function(Y,Z,Z_a,Z_b,Zobs_id,minr,minc,...){
   rtest_out = list(tobs=tobs,tvals=tvals)
   decision = out_pval(rtest_out,ret_pval,alpha=0.05)
 
+  valid_clique <- apply(conditional_clique, 2, function(x) length(unique(x))==1)
+  if (sum(valid_clique)!=0){
+    cat("The biclique containing Zobs is not valid \n")
```
Comment (dpuelz) [inline R/functions.R:128]: Could you change the error here from "valid" to something a bit more descriptive, like "the biclique containing Zobs contains only one exposure and will therefore result is a powerless test".  Just don't want to confuse valid with statistical validity.

File: R/functions.R [hunk 2]
```diff
@@ -103,8 +123,18 @@ clique_test = function(Y,Z,Z_a,Z_b,Zobs_id,minr,minc,...){
   rtest_out = list(tobs=tobs,tvals=tvals)
   decision = out_pval(rtest_out,ret_pval,alpha=0.05)
 
+  valid_clique <- apply(conditional_clique, 2, function(x) length(unique(x))==1)
```
Comment (dpuelz) [inline R/functions.R:126]: see below comment.  probably should change the function name too.

File: R/functions.R [hunk 3]
```diff
@@ -368,6 +398,115 @@ out_clique_decomposition = function(NEgraph,Zobs_id,minr,minc,stop_at_Zobs=TRUE)
...
+#'
+#' @param ne The null-exposure graph
+#' @param Zobs_id The index location of the observed assignment vector in \code{Z}, \code{Z_a}, and \code{Z_b}.
+#' @param num_ass Controls the minimum number of focal assignments in the biclique found
+#' by the algorithm (algorithm runtime is sensitive to this value).
+#' @param stop_at_Zobs A Boolean indicating whether the decomposition algorithm should stop when the clique containing the observed assignment is found.
+#' @return If \code{stop_at_Zobs} is \code{TRUE}, a matrix representing the clique to condition upon.
+#' If \code{stop_at_Zobs} is \code{FALSE}, a list containing the clique decomposition of the null-exposure graph.
+#' @export
+cliquedecom_greedy = function(ne, Zobs_id, num_ass, stop_at_Zobs){
```
Comment (dpuelz) [inline R/functions.R:473]: should we have "ne" be changed to "NEgraph" so that it matches the inputs to the other decom algo

**Review by dpuelz — CHANGES_REQUESTED** [review]
see greedy decom input naming below

File: R/functions.R (hunk 1 above)
Comment (szhuang1) [reply R/functions.R]: I've changed "valid" to "only one exposure for some focal assignment, and therefore the result is a powerless test", and change the corresponding return of the function.

File: R/functions.R (hunk 2 above)
Comment (szhuang1) [reply R/functions.R]: I've changed the variable name "valid_clique" to "check_clique_unique".

File: R/functions.R (hunk 3 above)
Comment (szhuang1) [reply R/functions.R]: I've changed all "ne" to "NEgraph".

Comment (szhuang1) [reply R/functions.R]: Also unified the names of greedy algo and bimax algo, in the format of "out_clique_decomposition_bimax/greedy". Also changed the greedy decom to "out_greedy_decom"

**Pull request merged**